packages/oauth/oauth-client/src/constants.ts

```diff
@@ -2,3 +2,20 @@
  * Per ATProto spec (OpenID uses RS256)
  */
 export const FALLBACK_ALG = 'ES256'
+
+/**
+ * A variable that allows to determine if unsecure origins should be allowed
+ * in OAuth related URI's. This variable is only set to `true` when NODE_ENV
+ * is either `development` or `test`.
+ */
+export const ALLOW_UNSECURE_ORIGINS = (() => {
+  // try/catch to support running in a browser, including when process.env is
+  // shimmed (e.g. by webpack)
+  try {
+    return (
+      process.env.NODE_ENV === 'development' || process.env.NODE_ENV === 'test'
+    )
+  } catch {
+    return false
+  }
+})()
```

Comment:
Is there a reason why we should set default based on `NODE_ENV` rather than allowing deployments to configure this themselves based on their particular situation?

Reply:
Ah, I see now, this convention already existed 👍  I usually lean towards having uniform behavior across `NODE_ENV`, or only treating `NODE_ENV=production` specially if anything.  Then testing or dev environments would manually set `allowUnsecure` themselves.  This makes the differences in environments explicit, and I think tends to produce fewer surprises when transitioning to run code in production.  Even so, we should probably not change how this works in this PR.

Reply:
You're right. I actually thought about doing this after I submitted for review.